pkg/reconciler/parallel/resources/subscription.go

```diff
@@ -69,10 +70,8 @@ func NewFilterSubscription(branchNumber int, p *v1.Parallel) *messagingv1.Subscr
 			},
 		}
 	} else {
-		r.Spec.Subscriber = &duckv1.Destination{
-			Ref: p.Spec.Branches[branchNumber].Filter.Ref,
-			URI: p.Spec.Branches[branchNumber].Filter.URI,
-		}
+		r.Spec.Subscriber = p.Spec.Branches[branchNumber].Filter.DeepCopy()
```

Comment:
Why we are making this change here?

Reply:
`Destination` has other fields beyond `URI` and `Ref` and we need to set them all, just better to assign the full copied struct instead of individual fields